proof/archive.go

```diff
@@ -140,6 +140,89 @@ type NotifyArchiver interface {
 	fn.EventPublisher[Blob, []*Locator]
 }
 
+// MultiArchiveNotifier is a NotifyArchiver that wraps several other archives
+// and notifies subscribers about new proofs that are added to any of the
+// archives.
+type MultiArchiveNotifier struct {
```

Comment:
Why do we need this in addition to `MultiArchiver`? `MultiArchiver` supports the notification interface.

Reply:
This just implements the reduced `NotifyArchiver` so we can support the multiverse store as an archive backend. If we wanted to use the existing `MultiArchiver`, we would need to implement the other methods from the `Archiver` interface in the multiverse DB.